whatshap/vcf.py

```diff
@@ -964,7 +987,17 @@ def write(self, chromosome: str, sample_superreads, sample_components):
                         is_het = not genotypes[pos].is_homozygous()
 
                     if pos in components and pos in phases and is_het:
-                        self._set_phasing_tags(call, components[pos], phases[pos])
+                        use_haploid_components = (
+                            haploid_components
+                            and pos in haploid_components
+                            and len(haploid_components[pos]) == self.ploidy
+                        )
+                        if use_haploid_components:
+                            self._set_phasing_tags(
+                                call, components[pos], phases[pos], haploid_components[pos]
+                            )
+                        else:
+                            self._set_phasing_tags(call, components[pos], phases[pos], None)
```

Comment:
This is what I meant (avoids writing out two calls to `self._set_phasing_tags`):
```suggestion
                        haploid_component = haploid_components[pos] if (
                            haploid_components
                            and pos in haploid_components
                            and len(haploid_components[pos]) == self.ploidy
                        ) else None
                        self._set_phasing_tags(
                            call, components[pos], phases[pos], haploid_component
                        )
```

Reply:
Ah, sorry I was not sure what you meant with the temporary variable. I pushed another version (and squashed it with the previous commit).

Reply:
Thanks, looks good now. Entirely my fault of not conveying what I wanted from the beginning :-).

(The reformatting that black has done now makes me wonder whether it would look better with an explicit `if (...): haploid_component = ...; else: haploid_component = None`, that is, not with an if expression, but let’s just keep this version now.)